src/rgw/rgw_process.cc

```diff
@@ -262,6 +263,42 @@ int rgw_process_authenticated(RGWHandler_REST * const handler,
   return 0;
 }
 
+int do_standard_bucket_logging(rgw::sal::Driver* driver,
+    req_state* s, 
+    RGWOp* op, 
+    optional_yield y) {
+  if (!s->bucket) {
+    // not a bucket operation
+    return 0;
+  }
+  // check if bucket logging is needed
+  const auto& bucket_attrs = s->bucket->get_attrs();
+  auto iter = bucket_attrs.find(RGW_ATTR_BUCKET_LOGGING); 
+  if (iter == bucket_attrs.end()) {
+    return 0;
+  }
+  rgw::bucketlogging::configuration configuration;
+  try {
+    configuration.enabled = true;
```

Comment:
you are right. but in AWS there is an object called `BucketLoggingStatus` in case it is empty, it means that bucket logging is disabled (or never existed). and in case it has bucket logging, it contains an object called `LoggingEnabled` where the actual configuration goes.
so, to reflect that in our structures i added this flag.
in case of decoding from XML (the REST API) this value is set based on the XML tags. but when decoding from our internal storage, i set the parameter based on the existence of the attribute.

since this boolean is only used when encoding/decoding XML, it might not be necessary to set it here (as well as in rgw_op.cc). but then you have a valid object with a flag indicating an incorrect state, which might eb confusing (e.g. when using gdb).